src/mlpack/methods/grad_boosting/grad_boosting.hpp

```diff
@@ -0,0 +1,237 @@
+/**
+ * @file methods/grad_boosting/grad_boosting.hpp
+ * @author Abhimanyu Dayal
+ *
+ * Gradient Boosting class. 
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+// Include guard - Used to prevent including double copies of the same code
+#ifndef MLPACK_METHODS_GRADBOOSTING_GRADBOOSTING_HPP
+#define MLPACK_METHODS_GRADBOOSTING_GRADBOOSTING_HPP
+
+// Importing base components required to write mlpack methods.
+#include <mlpack/core.hpp>
+
+// Only using decision trees for now, therefore only including decision tree functionalities.
+#include <mlpack/methods/decision_tree/decision_tree.hpp>
+
+// Written in mlpack namespace.
+namespace mlpack {
+/**
+ * The Gradient Boosting class. Gradient Boosting is a boosting algorithm, meaning that it
+ * combines an ensemble of weak learners to produce a strong learner.
+ * 
+ * Gradient Boosting is generally implemented using Decision Trees, or more specifically 
+ * Decision Stumps i.e. weak learner Decision Trees with low depth.
+ * 
+ * @tparam MatType Data matrix type (i.e. arma::mat or arma::sp_mat).
+ */
+
+// 
+template<
+  typename WeakLearnerType = ID3DecisionStump, 
+  typename MatType = arma::mat>
+class GradBoosting 
+{
+ public: 
+
+  typedef typename MatType::elem_type ElemType;
+
+  /**
+   * Constructor for creating GradBoosting without training. 
+   * Be sure to call Train() before calling Classify()
+   */
+  GradBoosting();
+
+  /**
+   * Constructor for a GradBoosting model. Any extra parameters are used as
+   * hyperparameters for the weak learner. These should be the last arguments
+   * to the weak learner's constructor or `Train()` function (i.e. anything
+   * after `numClasses` or `weights`).
+   *
+   * @param data Input data.
+   * @param labels Corresponding labels.
+   * @param numClasses The number of classes.
+   * @param numModels Number of weak learners.
+   * @param weakLearnerParams... Any hyperparameters for the weak learner.
+   */
+  template<typename... WeakLearnerArgs>
+  GradBoosting(
+    const MatType& data,
+    const arma::Row<size_t>& labels,
+    const size_t numClasses,
+    const size_t numModels = 10,
+    WeakLearnerArgs&&... weakLearnerArgs
```

Comment:
Please, put the first function parameter on the same line, and do the same for all functions